x-pack/plugins/monitoring_collection/server/lib/get_kibana_stats.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { ServiceStatus, ServiceStatusLevels } from '../../../../../src/core/server';
+
+const SNAPSHOT_REGEX = /-snapshot/i;
+
+const ServiceStatusToLegacyState: Record<string, string> = {
+  [ServiceStatusLevels.critical.toString()]: 'red',
+  [ServiceStatusLevels.unavailable.toString()]: 'red',
+  [ServiceStatusLevels.degraded.toString()]: 'yellow',
+  [ServiceStatusLevels.available.toString()]: 'green',
+};
+
+export async function getKibanaStats({
+  config,
+  overallStatus$,
+}: {
+  config: {
+    allowAnonymous: boolean;
+    kibanaIndex: string;
+    kibanaVersion: string;
+    uuid: string;
+    server: {
+      name: string;
+      hostname: string;
+      port: number;
+    };
+  };
+  overallStatus$: Observable<ServiceStatus>;
+}) {
+  const overallStatus = await overallStatus$.pipe(first()).toPromise();
+  return {
+    uuid: config.uuid,
+    name: config.server.name,
+    index: config.kibanaIndex,
+    host: config.server.hostname,
+    locale: i18n.getLocale(),
+    transport_address: `${config.server.hostname}:${config.server.port}`,
+    version: config.kibanaVersion.replace(SNAPSHOT_REGEX, ''),
+    snapshot: SNAPSHOT_REGEX.test(config.kibanaVersion),
+    status: ServiceStatusToLegacyState[overallStatus.level.toString()],
+  };
```

Comment:
Is the intention for plugins to use this as a model? In other words, use local data sources and provide metrics that are specific to the current Kibana instance?

Or, would a plugin like Alerting and Reporting provide metrics of their feature across the cluster? For example, should Reporting measure how many reports failed by looking at the history of all reports, or just keep a counter of the reports that failed in that instance?

Reply:
It's a great question. The current thinking is for the metrics to represent a single instance of Kibana exclusively, as we don't have a solution for cluster-wide Kibana-specific metrics. Elasticsearch can solve this by simply only exposing cluster-wide metrics through the master node but since Kibana has no such concept, we have no real way of avoiding duplication (by all Kibanas reporting the same cluster-level metrics at each collection period). Would love to hear any opposing or other thoughts here though!

Reply:
> The current thinking is for the metrics to represent a single instance of Kibana exclusively

Great! I really think this is the right way to go. I'm excited about this because I have drafted a PR that adds an HTTP endpoint for "reporting stats" for the local instance. It does it by paying attention to the "events" that get logged: https://github.com/elastic/kibana/pull/126968

Reply:
I did something similar with the actions and alerting plugins that track successful and failed executions: https://github.com/elastic/kibana/pull/123416/files#diff-ad0041a679432ee6b23beeaf52b621f07ab2ec7d80631cc66f6d590b0f58f0d2